src/java.desktop/unix/native/libawt_xawt/awt/awt_GraphicsEnv.c

```diff
@@ -572,6 +572,11 @@ getAllConfigs (JNIEnv *env, int screen, AwtScreenDataPtr screenDataPtr) {
 
 cleanup:
     if (success != JNI_TRUE) {
+        for (int i = 0; i < screenDataPtr->numConfigs; i++) {
+            if (graphicsConfigs[i] != 0) {
+                free(graphicsConfigs[i]);
+            }
```

Comment:
It seems  `screenDataPtr->numConfigs` is updated only when `success = JNI_TRUE` and here we are checking if `success != JNI_TRUE` condition, so amnot sure if we will be getting the correct updated configs number.
I think condition should be `i< ind-1` which is updated before every allocation, but it should be verified.

Reply:
I think "nConfig" would have been the better value to use.
Look at how the array is allocated and that it is zeroed out
   nConfig = n8p + n12p + n8s + n8gs + n8sg  + n1sg + nTrue + 1;
    graphicsConfigs = (AwtGraphicsConfigDataPtr *)
        calloc(nConfig, sizeof(AwtGraphicsConfigDataPtr));

So it should always be safe.